### PR TITLE
#166240784 Enables users  to create new article 

### PR DESCRIPTION
**Pull request by truestbyheart**
#### What does this PR do?
* It enables the user to create new articles

#### Description of Task to be completed?
 * Create a UI page for creating article
#### How should this be manually tested?
* Clone the repo to the local machine.
* Run ``npm install``   to install the dependencies.
* Set up the   ``.env`` as instructed by the   ``.env-example``
* Run  ``npm start``.
* Send a GET request to ``/new``.
#### What are the relevant pivotal tracker stories?
#[166240784](https://www.pivotaltracker.com/story/show/166240784)

### CODE REVIEW

File: src/Redux/Actions/articles.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+};
+
+export const getTags = () => (dispatch) => {
+  axios.get('/api/tags').then((res) => {
+    dispatch({ type: GET_TAGS, payload: res.data.tags });
+  });
+};
+export const changeState = data => (dispatch) => {
+  dispatch({ type: UPDATE_ARTICLE_STATE, payload: data });
+};
```
Comment (bdushimi) [inline src/Redux/Actions/articles.js:69]: @truestbyheart, The thunk name is not specific enough to reflect what it does actually! The thunk says changeState but the dispatched action is to update article state. Additionally, I think there's no need to add _STATE suffix, at the end of the day a dispatched action will always update the state.

File: src/Redux/Reducers/createArticle.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      };
+
+    default:
+      return state;
+  }
+};
+
+export const publish = (state = {}, action) => {
+  switch (action.type) {
+    case PUBLISH_ARTICLE:
```
Comment (bdushimi) [inline src/Redux/Reducers/createArticle.js:38]: @truestbyheart wouldn't be better to export one variable and have 1 switch statement block instead of two?

**Review by bdushimi — CHANGES_REQUESTED** [review]

File: src/App/_tests_/createArticle.test.js [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+  });
+});
+
+describe('Actions', () => {
+  test('create first draft', () => {
+    const expectedData = {
+      data: {
+        status: 201,
+        message: 'Article Created successfully',
+        article: { body: 'Can i get an Amen' }
```
Comment (MCFrank16) [inline src/App/_tests_/createArticle.test.js:147]: boy you need the holy spirit for sure!!!

File: src/App/Components/CreateArticle/CreateArticle.js [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+              });
+              toast.success('Article saved successfully');
+            })
+            .catch(() => {
+              const { errorMessage } = this.props;
+              toast.warn(errorMessage);
+            });
+        } else {
+          autosaveArticle(article)
+            .then(() => {
```
Comment (nkpremices) [inline src/App/Components/CreateArticle/CreateArticle.js]: I like this idea @truestbyheart

**Review by nkpremices — APPROVED** [review]
Good job man

File: src/App/_tests_/setup/setupEnzyme.js [hunk 5]
```diff
@@ -0,0 +1,4 @@
+import Enzyme from 'enzyme';
```
Comment (alainburindi) [inline src/App/_tests_/setup/setupEnzyme.js]: we don't need this file

File: public/index.html [hunk 6]
```diff
@@ -1,48 +1,37 @@
...
-</noscript>
-
-<div id="root"></div>
+  </noscript>
 
+  <div id="root"></div>
 </body>
-</html>
+
+</html>
```
Comment (alainburindi) [inline public/index.html:37]: new line

**Review by alainburindi — APPROVED** [review]

**Review by Ramadhan0 — APPROVED** [review]

**Pull request merged**